dot/sync/disjoint_block_set.go

```diff
@@ -80,13 +103,42 @@ type disjointBlockSet struct {
 
 	// map of parent hash -> child hashes
 	parentToChildren map[common.Hash]map[common.Hash]struct{}
+
+	timeNow func() time.Time
 }
 
 func newDisjointBlockSet(limit int) *disjointBlockSet {
 	return &disjointBlockSet{
 		blocks:           make(map[common.Hash]*pendingBlock),
 		parentToChildren: make(map[common.Hash]map[common.Hash]struct{}),
 		limit:            limit,
+		timeNow:          time.Now,
+	}
+}
+
+func (s *disjointBlockSet) start(ctx context.Context, done chan<- struct{}) {
```

Comment:
You should call that `run`. In my opinion `start` is something launching a goroutine (only starting something then returning), `run` on the other hand is 'running' aka blocking. Feel free to correct me or introduce a new word though, I'd be happy to use something else than `run`